### DISCUSSION TITLE
Unable to send 1M uluna with solid-vite example, on TerraClassic chain

### DISCUSSION THREAD

**Discussion by JeromeTGH** [coinhall/cosmes#66] [Q&A] [answered]
Hello,

When we use the solid-vite example, and when we select :
- "Terra Classic" as chain
- "Terra Station" as wallet
- "Extension" as wallet type

The broadcastTx function works well.

But, if we modify the amount in MsgSend, from "1" (i.e. 1 uluna / 0.000001 LUNC) to "1000000" (i.e. 1M uluna / 1 LUNC), the broadcastTx function return an error.

```
App.tsx:258 Error: insufficient fees; got: "2616675uluna", required: "87761uaud,87761ucad,64666uchf,452662ucny,415710udkk,57738ueur,50809ugbp,540423uhkd,1006942000uidr,5025472uinr,7561303ujpy,78523000ukrw,2621664uluna,197956945umnt,277140umyr,577375unok,3510440uphp,48471usdr,577375usek,92380usgd,2133978uthb,1847600utwd,69285uusd" = "87761uaud,87761ucad,64666uchf,452662ucny,415710udkk,57738ueur,50809ugbp,540423uhkd,1006942000uidr,5025472uinr,7561303ujpy,78523000ukrw,2616664uluna,197956945umnt,277140umyr,577375unok,3510440uphp,48471usdr,577375usek,92380usgd,2133978uthb,1847600utwd,69285uusd"(gas) + "5000uluna"(stability): insufficient fee
    at RpcClient.broadcastTx (RpcClient.ts:128:13)
    at async KeplrExtension.broadcastTx (ConnectedWallet.ts:158:18)
    at async HTMLButtonElement.broadcastTx [as $$click] (App.tsx:246:22)
```

It seems that the pb is related to the TerraClassic on-chain tax (0.5% of sended amount), which is in addition to gas costs.

**Comment by AaronCQL** [comment, marked as answer]
Hey @JeromeTGH, I've converted this issue into a discussion as it's not inherently an issue with CosmES, but with how Terra Classic handles fees. 

Since Terra Classic has a burn tax of 0.5%, you'll need to add the `5000uluna` (0.5% of the `1000000uluna` you are trying to send) into the gas fees that was estimated:

https://github.com/coinhall/cosmes/blob/78770cef4eba519a4372ebde34f09b1f3e0e0ada/examples/solid-vite/src/App.tsx#L243

This is because the fee estimation in Terra Classic does NOT take into consideration the burn tax.

**Comment by JeromeTGH** [comment]
Okay, thanks @AaronCQL !

I solved this problem as follows (in manual testing)

```
const fee = await wallet.estimateFee(tx, 1.8); // Note 2: feeMultiplier increased to 1.8, to compensate the "out of gas" error
const burnTax = new Coin({
  denom: getDenom(chain()),
  amount: "5000",                              // Note 1 : 5000 uluna added, for manual testing, corresponding to 0,5% of 1M uluna to send
});
fee.amount.push(burnTax);
console.log("Tx fee:", fee);
```

Perhaps can we complete your estimateFee function, with one more optional parameter (feeToAdd, for example).

What do you think about that ?
Thx !

**Reply by AaronCQL** [reply]
Yeap, this is similar to what we do as well on coinhall.org. Adding that additional parameter sounds useful, we'll look into it!
